<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Kaapi: Util::LogicError Class Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.6.3 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul>
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
      <li><a href="modules.html"><span>Modules</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li class="current"><a href="annotated.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
    </ul>
  </div>
  <div class="tabs">
    <ul>
      <li><a href="annotated.html"><span>Class&nbsp;List</span></a></li>
      <li><a href="classes.html"><span>Class&nbsp;Index</span></a></li>
      <li><a href="hierarchy.html"><span>Class&nbsp;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&nbsp;Members</span></a></li>
    </ul>
  </div>
  <div class="navpath"><a class="el" href="namespace_util.html">Util</a>::<a class="el" href="class_util_1_1_logic_error.html">LogicError</a>
  </div>
</div>
<div class="contents">
<h1>Util::LogicError Class Reference<br/>
<small>
[<a class="el" href="group___exception.html">Exception</a>]</small>
</h1><!-- doxytag: class="Util::LogicError" --><!-- doxytag: inherits="Util::Exception" -->
<p><code>#include &lt;utils_error.h&gt;</code></p>
<div class="dynheader">
Inheritance diagram for Util::LogicError:</div>
<div class="dynsection">
<div class="center"><img src="class_util_1_1_logic_error__inherit__graph.png" border="0" usemap="#_util_1_1_logic_error_inherit__map" alt="Inheritance graph"/></div>
<map name="_util_1_1_logic_error_inherit__map" id="_util_1_1_logic_error_inherit__map">
<area shape="rect" id="node5" href="class_util_1_1_invalid_argument_error.html" title="Util::InvalidArgumentError" alt="" coords="352,32,560,61"/><area shape="rect" id="node11" href="class_util_1_1_length_error.html" title="Util::LengthError" alt="" coords="387,85,525,115"/><area shape="rect" id="node13" href="class_util_1_1_no_more_resource.html" title="Util::NoMoreResource" alt="" coords="369,139,543,168"/><area shape="rect" id="node17" href="class_util_1_1_out_of_range_error.html" title="Util::OutOfRangeError" alt="" coords="367,192,545,221"/><area shape="rect" id="node2" href="class_util_1_1_exception.html" title="Base class for the KAAPI&#39;s exception hierarchy Base class for hierarchy of error..." alt="" coords="5,112,128,141"/><area shape="rect" id="node7" href="class_util_1_1_already_bind.html" title="Util::AlreadyBind" alt="" coords="608,5,749,35"/><area shape="rect" id="node9" href="class_util_1_1_no_found.html" title="Util::NoFound" alt="" coords="620,59,737,88"/><area shape="rect" id="node15" href="class_util_1_1_bad_alloc.html" title="Util::BadAlloc" alt="" coords="621,139,736,168"/></map>
<center><span class="legend">[<a href="graph_legend.html">legend</a>]</span></center></div>
<div class="dynheader">
Collaboration diagram for Util::LogicError:</div>
<div class="dynsection">
<div class="center"><img src="class_util_1_1_logic_error__coll__graph.png" border="0" usemap="#_util_1_1_logic_error_coll__map" alt="Collaboration graph"/></div>
<map name="_util_1_1_logic_error_coll__map" id="_util_1_1_logic_error_coll__map">
<area shape="rect" id="node2" href="class_util_1_1_exception.html" title="Base class for the KAAPI&#39;s exception hierarchy Base class for hierarchy of error..." alt="" coords="7,5,129,35"/></map>
<center><span class="legend">[<a href="graph_legend.html">legend</a>]</span></center></div>

<p><a href="class_util_1_1_logic_error-members.html">List of all members.</a></p>
<table border="0" cellpadding="0" cellspacing="0">
<tr><td colspan="2"><h2>Public Member Functions</h2></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_util_1_1_logic_error.html#aaa8c7614ed9428ac50adeb181eba3ec7">LogicError</a> (const std::string &amp;msg, int code=0)</td></tr>
<tr><td class="memItemLeft" align="right" valign="top">const char *&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_util_1_1_logic_error.html#a42a4dea50319d946372a9b8dc02f52e1">what</a> () const </td></tr>
<tr><td colspan="2"><h2>Protected Attributes</h2></td></tr>
<tr><td class="memItemLeft" align="right" valign="top">std::string&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_util_1_1_logic_error.html#a05ed4e18e87d856ef9384c6043610e7c">_msg</a></td></tr>
</table>
<hr/><a name="_details"></a><h2>Detailed Description</h2>
<p>Logic. An object of this class is throwed for error that could be, in principle, detected at compilation. </p>
<hr/><h2>Constructor &amp; Destructor Documentation</h2>
<a class="anchor" id="aaa8c7614ed9428ac50adeb181eba3ec7"></a><!-- doxytag: member="Util::LogicError::LogicError" ref="aaa8c7614ed9428ac50adeb181eba3ec7" args="(const std::string &amp;msg, int code=0)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">Util::LogicError::LogicError </td>
          <td>(</td>
          <td class="paramtype">const std::string &amp;&nbsp;</td>
          <td class="paramname"> <em>msg</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&nbsp;</td>
          <td class="paramname"> <em>code</em> = <code>0</code></td><td>&nbsp;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td><td></td>
        </tr>
      </table>
</div>
<div class="memdoc">

</div>
</div>
<hr/><h2>Member Function Documentation</h2>
<a class="anchor" id="a42a4dea50319d946372a9b8dc02f52e1"></a><!-- doxytag: member="Util::LogicError::what" ref="a42a4dea50319d946372a9b8dc02f52e1" args="() const " -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">const char* Util::LogicError::what </td>
          <td>(</td>
          <td class="paramname"></td>
          <td>&nbsp;)&nbsp;</td>
          <td> const<code> [virtual]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">
<p>Return an human readable string for the error message </p>

<p>Reimplemented from <a class="el" href="class_util_1_1_exception.html#ad15cb00bb04746ae283c3ddd20ebb9c6">Util::Exception</a>.</p>

</div>
</div>
<hr/><h2>Member Data Documentation</h2>
<a class="anchor" id="a05ed4e18e87d856ef9384c6043610e7c"></a><!-- doxytag: member="Util::LogicError::_msg" ref="a05ed4e18e87d856ef9384c6043610e7c" args="" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">std::string <a class="el" href="class_util_1_1_logic_error.html#a05ed4e18e87d856ef9384c6043610e7c">Util::LogicError::_msg</a><code> [protected]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

</div>
</div>
<hr/>The documentation for this class was generated from the following file:<ul>
<li><a class="el" href="utils__error_8h.html">utils_error.h</a></li>
</ul>
</div>
<hr class="footer"/><address style="text-align: right;"><small>Generated on Wed Jun 16 08:53:23 2010 for Kaapi by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.6.3 </small></address>
</body>
</html>
